flake.nix

```diff
@@ -0,0 +1,169 @@
+{
+  description = "Shadysim";
+
+  inputs = {
+
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-21.05";
+
+    pycrypto-src = {
+      url = "github:pycrypto/pycrypto";
+      flake = false;
+    };
+
+    simtools-src = {
+      url = "git+https://git.osmocom.org/sim/sim-tools";
+      flake = false;
+    };
+
+    imsi-pseudo-src = {
+      url = "git+https://git.osmocom.org/imsi-pseudo";
+      flake = false;
+    };
+
+  };
+
+
+  outputs =
+    { self
+    , nixpkgs
+    , pycrypto-src
+    , simtools-src
+    , imsi-pseudo-src
+    , ...
+    }:
+    let
+      supportedSystems = [ "x86_64-linux" "x86_64-darwin" ];
+      forAllSystems = f: nixpkgs.lib.genAttrs supportedSystems (system: f system);
+
+      nixpkgsFor = forAllSystems (system:
+        import nixpkgs {
+          inherit system;
+          overlays = [ self.overlay ];
+        }
+      );
+
+    in
+    {
+
+      overlay = final: prev: rec {
+
+        pycrypto =
+          with final.python2Packages;
+          buildPythonPackage {
+            pname = "pycrypto";
+            version = "2.6.1";
+            src = pycrypto-src;
+          };
+
+        shadysim =
+          { python2
+          , python2Packages
+          }:
+
+            with python2Packages;
+            buildPythonApplication {
+              name = "shadysim";
+              version = "0.1.0";
+              src = "${simtools-src}/shadysim";
+
+              patches = [
+                ./patches/0001-add-setup-py.patch
```

Comment:
this patch is just adding a file with some contents.
Rather than adding a patch could you just 
```
{
echo "..."
echo "..."
echo "..."
} > setup.py
```

patches are when you need to patch the source, it's a little clearer I find in a preConfigure or postConfigure phase to just add the file. (if there are no conficts)

Reply:
I think its better to add a real file to the repo, then copy it via:
```nix
postPatch = "cp ${./files/setup.py}" ./setup.py"
```

Reply:
>`postPatch = "cp ${./files/setup.py}" ./setup.py"`

Yeah I think I like this more, @happysalada what do you think?

Reply:
it's good too.
The only downside is that it forces people to look into another file. This is why I like the echo approach better for short files. You can see everything right then and there.

Reply:
Hmm yeah, I like a flat file structure too, I will go with this then.